src/connectors/rating-menu/connectRatingMenu.ts

```diff
@@ -392,10 +399,12 @@ const connectRatingMenu: RatingMenuConnector = function connectRatingMenu(
         }
         facetValues = facetValues.reverse();
 
+        const hasNoResults = results ? results.nbHits === 0 : true;
+
         return {
           items: facetValues,
-          hasNoResults: results ? results.nbHits === 0 : true,
-          canRefine: facetValues.length > 0,
+          hasNoResults,
+          canRefine: (!hasNoResults || refinementIsApplied) && totalCount > 0,
```

Comment:
Can you walk me through the idea behind the addition of `totalCount > 0`?

From what I understand now, this conflicts with `(!hasNoResults || refinementIsApplied)` because if `hasNoResults` is truthy that means that there are no results (hits), which also means the count of facet values will amount to 0.

Reply:
Let's say the user first clicks on 4 stars and above cause they want to see good movies, they can type a query which will yield no results but still have a total facet value which is more than 0, [like in this example](https://owgvsc.csb.app/?movies%5Bquery%5D=starship%20troopers&movies%5BratingMenu%5D%5Brating%5D=4) (sorry it does not look obvious it's selected though)
Now if the user types some more stuff and the total facet count is 0 we know this query won't ever yield any result, and `canRefine` will be false as opposed to the previous implem.

Reply:
Thanks Aymeric, that's clearer with your example 👍 .